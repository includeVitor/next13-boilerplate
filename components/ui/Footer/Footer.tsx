'use client'

import { AppRoutes, AppSegments } from '@config/constants'
import Link from 'next/link'
import { useSelectedLayoutSegment } from 'next/navigation'

const Footer = () => {
    const segment = useSelectedLayoutSegment()

    if (segment === AppSegments.posts)
        return (
            <footer className="mt-12 text-blue-500 hover:underline text-lg">
                <Link href={AppRoutes.home}>← Back to home</Link>
            </footer>
        )

    return null
}

export default Footer
