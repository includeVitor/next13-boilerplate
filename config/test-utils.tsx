// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import { render, RenderOptions } from "@testing-library/react";
import { FC, ReactElement } from "react";

type ProvidersProps = {
    children: React.ReactNode;
};

const Providers: FC<ProvidersProps> = ({ children }) => {
    return <>{children}</>;
};

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, "queries">
) => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from "@testing-library/react";
// override render method
export { customRender as render };
