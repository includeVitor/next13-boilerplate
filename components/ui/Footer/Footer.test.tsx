import { render } from "@config/test-utils";
import Footer from "./Footer";

describe("Footer", () => {
    it("Should be able to match snapshot", () => {
        const { container } = render(<Footer />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
