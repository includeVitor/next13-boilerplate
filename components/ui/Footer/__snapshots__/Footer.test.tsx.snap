// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Should be able to render footer inside posts segment 1`] = `
<footer
  class="mt-12 text-blue-500 hover:underline text-lg"
>
  <a
    href="/"
  >
    ‚Üê Back to home
  </a>
</footer>
`;

exports[`Footer Should be able to render footer outside posts segment 1`] = `null`;
